### DISCUSSION TITLE
Unable to delete repository due to HttpError: Must have admin rights to Repository.

### DISCUSSION THREAD

**Discussion by lyqht** [octokit/octokit.js#2228] [Q&A] [answered]
Hello folks, I'm creating a project called [Unfork](https://github.com/lyqht/Unfork) for users to easily login using their GitHub account and delete forks that they have. Using the provider_token, I was able to get the repos that the user is an owner of.

```
const octokit = new Octokit({
  auth: provider_token,
});

await octokit.rest.repos.listForAuthenticatedUser();
```

<img width="570" alt="image" src="https://user-images.githubusercontent.com/35736525/174020921-3c3b21d3-fb28-4fea-b28d-49fa6bb0b6f7.png">

However, I'm getting an error when I try to **delete repos**.

```
error - HttpError: Must have admin rights to Repository.
```

This is the relevant code used 

```
const example = {
  owner: 'lyqht',
  repo: 'expo'
};
const response = await octokit.rest.repos.delete(example);
```

Any help will be appreciated!

**Comment by timrogers** [comment, marked as answer]
Hi there @lyqht! 👋 It sounds like the credential you're using to access the API doesn't have permission to delete a repo. How are you authenticating?

If you're not sure, then I'm happy to take a look at our logs and figure it out. To do that, I'll need the request ID. You can get that by catching the error and logging `error.response.headers`.

**Reply by lyqht** [reply]
hello tim, this is what i can find from the headers

`'x-oauth-client-id': '73ec5f0638230c1551ee'`
`'x-oauth-scopes': 'user:email'`

is it because the oauth scope is not including the delete repo scope? 

If it is because of that, how do i get this scope? I only made the API requests call using the provider token that I got from login with GitHub functionality using supabase 😆 Do I have to create a OAuth app?

**Reply by jaaneh** [reply]
> If it is because of that, how do i get this scope? I only made the API requests call using the provider token that I got from login with GitHub functionality using supabase 😆 Do I have to create a OAuth app?

Have you tried adding the necessary OAuth scopes to supabase? 
https://twitter.com/supabase/status/1369647270388523010
https://docs.github.com/en/developers/apps/building-oauth-apps/scopes-for-oauth-apps

**Reply by timrogers** [reply]
Yep! That should do the trick. You would need the `repo` scope to be able to delete a repo.

**Reply by gr2m** [reply]
I think you'll need the `delete_repo` scope which is separate

**Reply by timrogers** [reply]
> I think you'll need the `delete_repo` scope which is separate

Thanks for keeping me honest! I didn't realise we had a separate scope for that.

**Reply by lyqht** [reply]
hello all, thanks for the help! following @jaaneh's suggestion for oauth and @gr2m for the scope name, I was able to delete repos.
